test/Dialect/FIRRTL/imdce.mlir

```diff
@@ -561,3 +561,42 @@ firrtl.circuit "DeadPublic" {
      firrtl.instance pdc @PublicDeadChild()
   }
 }
+
+// -----
+// OMIR annotation should not block removal.
+//   - See: https://github.com/llvm/circt/issues/6199
+//
+// CHECK-LABEL: firrtl.circuit "OMIRRemoval"
+firrtl.circuit "OMIRRemoval" {
+  firrtl.module @OMIRRemoval() {
+    // CHECK-NOT %tmp_0
+    %tmp_0 = firrtl.wire {
+      annotations = [
+        {
+           class = "freechips.rocketchip.objectmodel.OMIRTracker",
+           id = 0 : i64,
+           type = "OMReferenceTarget"
+        }
+      ]} : !firrtl.uint<1>
+
+    // CHECK-NOT %tmp_1
+    %tmp_1 = firrtl.wire {
+      annotations = [
+        {
+           class = "freechips.rocketchip.objectmodel.OMIRTracker",
+           id = 1 : i64,
+           type = "OMMemberReferenceTarget"
+        }
+      ]} : !firrtl.uint<2>
+
+    // CHECK-NOT %tmp_2
+    %tmp_2 = firrtl.wire {
+      annotations = [
+        {
+           class = "freechips.rocketchip.objectmodel.OMIRTracker",
+           id = 3 : i64,
+           type = "OMMemberInstanceTarget"
+        }
+      ]} : !firrtl.uint<3>
+  }
```

Comment:
If not already, may be good to add examples where these should NOT be removed (and similarly other tests, hopefully can clone that around?).

Reply:
No examples of this. I added one to each. If I add in another annotation alongside this one, then removal is blocked. Good thinking.